Classes/Command/UnduplicateCommand.php

```diff
@@ -96,47 +118,91 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $this->output->title($this->getDescription());
 
         $this->dryRun = $input->getOption('dry-run');
+        $onlyThisIdentifier = $input->getOption('identifier');
+        $onlyThisStorage = (int) $input->getOption('storage') ;
 
         $queryBuilder = $this->connectionPool->getQueryBuilderForTable('sys_file');
-        $statement = $queryBuilder->count('*')
-            ->addSelect('identifier')
+        $queryBuilder->count('*')
+            ->addSelect('identifier', 'storage')
             ->from('sys_file')
-            ->groupBy('identifier')
-            ->having('COUNT(*) > 1')
+            ->groupBy('identifier', 'storage')
+            ->having('COUNT(*) > 1');
+        $whereExpressions = [];
+        if ($onlyThisIdentifier) {
+            $whereExpressions[] = $queryBuilder->expr()->eq(
+                    'identifier', $queryBuilder->createNamedParameter($onlyThisIdentifier, \PDO::PARAM_STR)
+            );
+        }
+        if ($onlyThisStorage > -1) {
+            $whereExpressions[] = $queryBuilder->expr()->eq(
+                    'storage', $queryBuilder->createNamedParameter($onlyThisStorage, Connection::PARAM_INT)
+            );
+        }
+        if ($whereExpressions) {
+            $queryBuilder->where(...$whereExpressions);
+        }
+        $statement = $queryBuilder
             ->execute();
 
-        while ($row = $statement->fetch()) {
-            $files = $this->findDuplicateFilesForIdentifier($row['identifier']);
+        while ($row = $statement->fetchAssociative()) {
```

Comment:
When no duplicates were found (`!$statement->rowCount()`), we should show a message to the user and we can stop the command before entering the while loop.

Reply:
I don't know if a message should be displayed if no duplicates found. I return different exit code now for dupes found and not found. See commit message.

Reply:
`$statement->rowCount()` has cross database backend issues / php dbdriver for SELECT. COUNT() is a select query. It may reporting wrong values or 0 even if records are found.

TYPO3 core had these issue some times so it's a good meant point to avoid using `rowCount()`.

See warning in TYPO3 documentation: https://docs.typo3.org/m/typo3/reference-coreapi/9.5/en-us/ApiOverview/Database/Statement/Index.html#rowcount

Reply:
@sbuerk Thank you. I fixed this now in PR #9: 

PR #9 is based in this PR and makes further changes so I think it is best to close here to avoid confusion. All changes in this PR are also in #9.